src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2722,9 +2725,8 @@ const bool java_bytecode_convert_methodt::is_method_inherited(
         return true;
       // methods with the default access modifier are only
       // accessible within the same package.
-      else if(access==ID_default &&
-              java_class_to_package(id2string(parent))==classpackage)
-        return true;
+      else if(access==ID_default)
+        return java_class_to_package(id2string(parent))==classpackage;
```

Comment:
Note that you would never have found a class in a different package, because the `else` case would have triggered an assertion failure.
Anyway, as you said the code has been rewritten, and thus this change is no longer necessary. Having multiple PRs fix the same problem hurts, but that's a different story. I'll close this PR.